### PR TITLE
Small test improvements

### PR DESCRIPTION
**Pull request by ghost**
This PR makes the determinism/non-determinism tests into property tests, and makes the `exact-integer-complex-generator` test more strict.

### CODE REVIEW

**Comment by ghost** [comment]
To clarify, this has no changes/effect on the SRFI itself.

File: property-test-tests.scm [hunk 1]
```diff
@@ -202,15 +204,19 @@
...
 
 (test-group "determinism"
   (parameterize ((current-random-source (make-random-source)))
-    (let ((lst1 (generator->list (boolean-generator) 1001)))
+    (let ((gen1 (gdrop (exact-number-generator) 30)))
       (parameterize ((current-random-source (make-random-source)))
-        (let ((lst2 (generator->list (boolean-generator) 1001)))
-          (test-assert (equal? lst1 lst2)))))))
+        (let ((gen2 (gdrop (exact-number-generator) 30)))
+          (test-property (lambda (x y)
```
Comment (arthurgleckler) [inline property-test-tests.scm]: Would you like to simplify this to `(test-property = (list gen1 gen2))`?

Comment (ghost) [reply property-test-tests.scm]: Yes that's a good idea. I don't have my development computer with me now, but I can fix that in a few hours. Or I have "Allow edits by maintainers" Github option checked if you wanted to make the change.

**Comment by arthurgleckler** [comment]
Thanks.  Done.  Merged and pushed.

**Pull request merged**